### PR TITLE
Fix `optuna.visualization.plot_contour` of subplot case with categorical axes.

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

See #2242

---

This PR's figure:

```python
import optuna

def objective(trial):
    trial.suggest_categorical("a", ["one", "two", "three"])
    trial.suggest_categorical("b", ["eins", "zwei", "drei"])
    trial.suggest_categorical("c", ["uno", "dos", "tres"])
    return 0

study = optuna.create_study()
study.optimize(objective, n_trials=50)

optuna.visualization.plot_contour(study)
```
newplot

### CODE REVIEW

**Comment by nzw0301** [comment]
Maybe we can remove `update_category_axes`?

**Comment by nzw0301** [comment]
Let me explain the comment above:

To set the categorical axis explicitly,  we need to evaluate this line with `True`
https://github.com/optuna/optuna/blob/8230e3e1c735f2f0e38c66345dcd6d681446e91f/optuna/visualization/_contour.py#L211
However, `update_category_axes` contains `True` value if the categorical value contains at least one numerical value in

https://github.com/optuna/optuna/blob/8230e3e1c735f2f0e38c66345dcd6d681446e91f/optuna/visualization/_contour.py#L140-L142

So I came up with replacing `update_category_axes` with `_is_categorical`. In this case, I think we do not need to `update_category_axes` anymore.

**Comment by HideakiImamura** [comment]
@keisuke-umezawa Could you review this PR if you have time?

**Comment by nzw0301** [comment]
@keisuke-umezawa What do you think? If https://github.com/optuna/optuna/pull/2297#issuecomment-771456414 is fine, I push additional commits to replace `update_category_axes` with `_is_categorical`, and update the test.

**Review by keisuke-umezawa — COMMENTED** [review]
> @keisuke-umezawa What do you think? If #2297 (comment) is fine, I push additional commits to replace update_category_axes with _is_categorical, and update the test.

If it passes all tests, it should be good.

File: optuna/visualization/_contour.py [hunk 1]
```diff
@@ -208,9 +208,9 @@ def _get_contour_plot(
                 figure.update_xaxes(range=param_values_range[x_param], row=y_i + 1, col=x_i + 1)
                 figure.update_yaxes(range=param_values_range[y_param], row=y_i + 1, col=x_i + 1)
 
-                if update_category_axes.get(x_param, False):
+                if _is_categorical(trials, x_param):
                     figure.update_xaxes(type="category", row=y_i + 1, col=x_i + 1)
-                if update_category_axes.get(y_param, False):
+                if _is_categorical(trials, y_param):
```
[the remark this answers is not in the document]
Comment (nzw0301) [reply optuna/visualization/_contour.py:210]: Yes. I have updated this PR for them and https://github.com/optuna/optuna/pull/2297#issuecomment-773036974 .

**Comment by nzw0301** [comment]
Thank you for your feedback! The test seems fine.

**Review by HideakiImamura — APPROVED** [review]
Thanks for your update. LGTM.

**Review by keisuke-umezawa — APPROVED** [review]
LGTM!

**Comment by nzw0301** [comment]
Thanks!

**Pull request merged**